include/marisa/key.h

```diff
@@ -49,6 +59,11 @@ class Key {
     union_.weight = weight;
   }
 
+#if __cplusplus >= 201703L
+  std::string_view str() const {
+    return ptr_ != NULL ? std::string_view(ptr_, length_) : std::string_view();
```

Comment:
Is `return std::string_view(nullptr, 0)` illegal?

Reply:
An answer from stack overflow had led me to believe construction with a nullptr wasn't legal, but after reading the standard, I now agree it is legal.